src/main/java/stirling/software/SPDF/config/security/CustomLogoutSuccessHandler.java

```diff
@@ -14,74 +14,56 @@
-            throws IOException, ServletException {
-
+            throws IOException {
         if (!response.isCommitted()) {
-            // Handle user logout due to disabled account
-            if (request.getParameter("userIsDisabled") != null) {
-                response.sendRedirect(
-                        request.getContextPath() + "/login?erroroauth=userIsDisabled");
-                return;
-            }
-            // Handle OAuth2 authentication error
-            if (request.getParameter("oauth2AuthenticationErrorWeb") != null) {
-                response.sendRedirect(
-                        request.getContextPath() + "/login?erroroauth=userAlreadyExistsWeb");
-                return;
-            }
             if (authentication != null) {
-                // Handle SAML2 logout redirection
                 if (authentication instanceof Saml2Authentication) {
+                    // Handle SAML2 logout redirection
                     getRedirect_saml2(request, response, authentication);
-                    return;
-                }
-                // Handle OAuth2 logout redirection
-                else if (authentication instanceof OAuth2AuthenticationToken) {
+                } else if (authentication instanceof OAuth2AuthenticationToken) {
+                    // Handle OAuth2 logout redirection
                     getRedirect_oauth2(request, response, authentication);
-                    return;
-                }
-                // Handle Username/Password logout
-                else if (authentication instanceof UsernamePasswordAuthenticationToken) {
-                    getRedirectStrategy().sendRedirect(request, response, "/login?logout=true");
-                    return;
-                }
-                // Handle unknown authentication types
-                else {
+                } else if (authentication instanceof UsernamePasswordAuthenticationToken) {
+                    // Handle Username/Password logout
+                    getRedirectStrategy().sendRedirect(request, response, LOGOUT_PATH);
+                } else {
+                    // Handle unknown authentication types
                     log.error(
-                            "authentication class unknown: "
-                                    + authentication.getClass().getSimpleName());
-                    getRedirectStrategy().sendRedirect(request, response, "/login?logout=true");
-                    return;
+                            "Authentication class unknown: {}",
+                            authentication.getClass().getSimpleName());
+                    getRedirectStrategy().sendRedirect(request, response, LOGOUT_PATH);
                 }
             } else {
                 // Redirect to login page after logout
-                getRedirectStrategy().sendRedirect(request, response, "/login?logout=true");
-                return;
+                getRedirectStrategy().sendRedirect(request, response, LOGOUT_PATH);
```

Comment:
With this logic, the user will only get the message `You have been logged out.`.

Better:

```java
String path = checkForErrors(request);

getRedirectStrategy().sendRedirect(request, response, path);

```